packages/contracts/contracts/router/modules/exchanges/NFTXModule.sol

```diff
@@ -0,0 +1,226 @@
+        bool isERC721 = collection.supportsInterface(ERC721_INTERFACE);
+        bool isERC1155 = collection.supportsInterface(ERC1155_INTERFACE);
+
+        if (isERC721) {
+            _approveERC721IfNeeded(IERC721(address(collection)), address(NFTX_MARKETPLACE));
+        } else if (isERC1155) {
+            _approveERC1155IfNeeded(IERC1155(address(collection)), address(NFTX_MARKETPLACE));
+        }
+
+        // Execute the fill
+        if (isERC721) {
+            try NFTX_MARKETPLACE.mintAndSell721WETH(
+                sellOrder.vaultId,
+                sellOrder.specificIds,
+                sellOrder.price,
+                sellOrder.path,
+                address(this)
+            ) {
+                // Pay fees
+                uint256 feesLength = fees.length;
+                for (uint256 i; i < feesLength; ) {
+                    Fee memory fee = fees[i];
+                    _sendERC20(fee.recipient, fee.amount, sellOrder.currency);
+                    unchecked {
+                        ++i;
+                    }
+                }
+                // Forward any left payment to the specified receiver
+                _sendAllERC20(receiver, sellOrder.currency);
+            } catch Error(string memory reason) {
+                emit LogStr(reason);
+                // Revert if specified
+                if (revertIfIncomplete) {
+                    revert UnsuccessfulFill();
+                }
+            }
+
+        } else if (isERC1155) { 
+            try NFTX_MARKETPLACE.mintAndSell1155WETH(
+                sellOrder.vaultId, 
+                sellOrder.specificIds,
+                sellOrder.amounts,
+                sellOrder.price,
+                sellOrder.path,
+                address(this)
+            ) {
+                // Pay fees
+                uint256 feesLength = fees.length;
+                for (uint256 i; i < feesLength; ) {
+                    Fee memory fee = fees[i];
+                    _sendERC20(fee.recipient, fee.amount, sellOrder.currency);
+                    unchecked {
+                        ++i;
+                    }
+                }
+
+                // Forward any left payment to the specified receiver
+                _sendAllERC20(receiver, sellOrder.currency);
+            } catch {
+                // Revert if specified
+                if (revertIfIncomplete) {
+                    revert UnsuccessfulFill();
+                }
+            }
+        }
+
+        uint256 length = sellOrder.specificIds.length;
+        for (uint256 i = 0; i < length; ) {
+             if (isERC721) {
+                _sendAllERC721(receiver, IERC721(address(collection)), sellOrder.specificIds[i]);
+            } else if (isERC1155) {
+                _sendAllERC1155(receiver, IERC1155(address(collection)), sellOrder.specificIds[i]);
+            }
+            unchecked {
+                ++i;
+            }
+        }
+       
+    }
+
+    // --- ERC721 hooks ---
```

Comment:
Since NFTX supports ERC1155, we should also support the ERC1155 on-received hook.